### PR TITLE
wifi: Fixes for RC1 issues

### PR DESCRIPTION
**Pull request by krish2718**
The driver still resides in Zephyr upstream, but docs are only in NCS as Zephyr doesn't have vedor docs.

- [x] Fix `options-from-kconfig` extensions to take a custom path @gmarull 
- [ ] Add `external_context` extensions to pull in RST + Doxygen docs from `modules/hal/nordic/drivers/nrf_wifi/doc` @gmarull

### CODE REVIEW

**Comment by krish2718** [comment]
Doc failure is due to #1

**Review by sachinthegreen — APPROVED** [review]

**Comment by gmarull** [comment]
> The driver still resides in Zephyr upstream, but docs are only in NCS as Zephyr doesn't have vedor docs.
> 
> * [ ]  Fix `options-from-kconfig` extensions to take a custom path @gmarull
> * [ ]  Add `external_context` extensions to pull in RST + Doxygen docs from `modules/hal/nordic/drivers/nrf_wifi/doc` @gmarull

For external_content simply extend `doc/nrf/conf.py` paths, and for Doxygen the same in `doc/nrf/nrf.doxyfile.in`.

**Comment by krish2718** [comment]
> > The driver still resides in Zephyr upstream, but docs are only in NCS as Zephyr doesn't have vedor docs.
> > 
> > * [ ]   Fix `options-from-kconfig` extensions to take a custom path @gmarull
> > * [ ]   Add `external_context` extensions to pull in RST + Doxygen docs from `modules/hal/nordic/drivers/nrf_wifi/doc` @gmarull
> 
> For external_content simply extend `doc/nrf/conf.py` paths, and for Doxygen the same in `doc/nrf/nrf.doxyfile.in`.

The `nrfx` (`hal_nordic`) docset from the utils has been removed https://github.com/nrfconnect/sdk-nrf/blame/main/doc/_utils/utils.py#L22 should I bring it back or just hardcode the path `HAL_NORDIC_BASE`?

File: doc/nrf/drivers/wifi/native/nrf70.rst [hunk 1]
```diff
@@ -87,14 +87,16 @@ Devicetree Specification configuration
...
 
 Multiple calibrations and checks are implemented to ensure consistency across channels and devices.
 However, these values have a dependency on PCB design, which may result in Error Vector Magnitude (EVM) and spectral mask failures.
 To avoid this problem, you can specify the power ceiling at which the EVM and spectral mask are met for a given PCB design.
 Additionally, build-time parameters are made available to drivers through the DTS overlay file.
 
-The following code snippet shows an example of the DTS overlay file:
+The following code snippet shows an example of the DTS overlay file. Note that the numbers filled in the following example do not represent any particular PCB design or package type.
+These are just example numbers. Customer should fill these values based on measurements obtained from transmitter testing on their PCB design. These numbers are represented in 0.25 dB steps.
+If 15 dBm needs to be configured then value 60 (0x3c in hex) should be configured.
```
Comment (bama-nordic) [inline doc/nrf/drivers/wifi/native/nrf70.rst:99]: I am suggesting a little re-wording below. Let's review and update -

The following code block shows an example of the DTS overlay file. The entries shown are only to serve as examples and don’t represent values for any particular PCB design or package type.
Customers are expected to fill in the appropriate values based on measurements obtained from transmitter testing on their custom product. The values are represented in 0.25 dB steps. For example, if 15 dBm needs to be configured then value 60 or 0x3c (hex) should be used.

File: doc/nrf/drivers/wifi/portable/nrf70.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.. nrf70_portable_wifi:
+
+nRF Wi-Fi portable driver
+#########################
+
+The nRF Wi-Fi portable driver implements OS agnostic code for the nRF70 FullMAC Series of devices.
```
Comment (richabp) [inline doc/nrf/drivers/wifi/portable/nrf70.rst:6]:
```suggestion
The nRF Wi-Fi portable driver implements OS-agnostic code for the nRF70 FullMAC Series of devices.
```

File: doc/nrf/drivers/wifi/portable/nrf70.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+This can be used to implement OS native drivers for the nRF70 Series devices.
```
Comment (richabp) [inline doc/nrf/drivers/wifi/portable/nrf70.rst:7]:
```suggestion
This code can be used to implement OS-native drivers for the nRF70 Series devices.
```

File: doc/nrf/drivers/wifi/native/nrf70.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+Devicetree Specification configuration
+======================================
+
+The maximum transmit power achieved on a nRF70 Series device-based product depends on the frequency band and operating channel.
+This varies over different :term:`Printed Circuit Board (PCB)` designs.
+
+Multiple calibrations and checks are implemented to ensure consistency across channels and devices.
+However, these values have a dependency on PCB design, which may result in Error Vector Magnitude (EVM) and spectral mask failures.
+To avoid this problem, you can specify the power ceiling at which the EVM and spectral mask are met for a given PCB design.
```
Comment (bama-nordic) [inline doc/nrf/drivers/wifi/native/nrf70.rst:94]: This ensures that the final maximum transmit power won't be above these specified values.

File: doc/nrf/drivers/wifi/portable/nrf70.rst [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+The Zephyr native driver implementation is located in the :file:`${ZEPHYR_BASE}/drivers/wifi/nrfwifi/` folder.
+
+API documentation
+*****************
+
+After the nRF Wi-Fi driver has been initialized, the application will see it as an Ethernet interface.
+To use the Ethernet interface, the application can use `Zephyr Network APIs`_.
```
Comment (richabp) [inline doc/nrf/drivers/wifi/portable/nrf70.rst:15]:
```suggestion
Once the nRF Wi-Fi driver is initialized, the application recognizes it as an Ethernet interface.
To use the Ethernet interface, the application can use the `Zephyr Network APIs`_.
```

File: doc/nrf/drivers/wifi/portable/nrf70.rst [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+See the :ref:`nrfx:nrf_wifi_api` to learn more about various modes of low-level API.
```
Comment (richabp) [inline doc/nrf/drivers/wifi/portable/nrf70.rst:17]:
```suggestion
See the :ref:`nrfx:nrf_wifi_api` to learn more about various modes of the low-level API.
```

**Review by richabp — CHANGES_REQUESTED** [review]
I think it requires an additional Wi-Fi driver top page, with the following subpages-

- nRF Wi-Fi driver 
- nRF Wi-Fi portable driver

**Comment by richabp** [comment]
Please remove the third bullet about the driver documentation form the "Removed" section of "Documentation" from the Changelog https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/releases/release-notes-changelog.html#documentation.

**Comment by krish2718** [comment]
> I think it requires an additional Wi-Fi driver top page, with the following subpages-
> 
> * nRF Wi-Fi driver
> * nRF Wi-Fi portable driver

we already have it https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-17653/nrf/drivers/wifi.html

**Comment by krish2718** [comment]
Only pending items it to link to the generated doxygen content.

**Review by bama-nordic — APPROVED** [review]

File: doc/nrf/drivers/wifi/nrf70_native.rst [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+   :alt: nRF Wi-Fi driver block diagram
+   :align: center
+   :figclass: align-center
+
+   nRF Wi-Fi driver architecture overview
+
+Design overview
+***************
+
+The nRF Wi-Fi driver follows an OS agnostic design, and the driver implementation is split into OS-agnostic and OS (Zephyr)-specific code.
```
Comment (richabp) [inline doc/nrf/drivers/wifi/nrf70_native.rst:36]:
```suggestion
The nRF Wi-Fi driver follows an OS-agnostic design, and the driver implementation is split into OS-agnostic and OS (Zephyr)-specific code.
```

File: doc/nrf/drivers/wifi/nrf70_native.rst [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+The maximum transmit power achieved on an nRF70 Series device-based product depends on the frequency band and operating channel.
+This varies across different :term:`Printed Circuit Board (PCB)` designs.
+
+Multiple calibrations and checks are implemented to ensure consistency across channels and devices.
+However, these values depend on PCB design, which may result in Error Vector Magnitude (EVM) and spectral mask failures.
+To avoid this problem, you can specify the power ceiling at which the EVM and spectral mask are met for a given PCB design.
+Additionally, build-time parameters are made available to drivers through the DTS overlay file.
+
+The following code snippet shows an example of the DTS overlay file.
+Note that the numbers filled in the following example do not represent any particular PCB design or package type.
```
Comment (richabp) [inline doc/nrf/drivers/wifi/nrf70_native.rst:100]: Consider converting this into note if "note" has to remain in the sentence.
```suggestion
Note that the numbers used in this example do not represent any particular PCB design or package type.
```

File: doc/nrf/drivers/wifi/native/nrf70.rst [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+			wifi-max-tx-pwr-2g-mcs7 = <0x40>;
+			wifi-max-tx-pwr-5g-low-mcs0 = <0x38>;
+			wifi-max-tx-pwr-5g-low-mcs7 = <0x38>;
+			wifi-max-tx-pwr-5g-mid-mcs0 = <0x38>;
+			wifi-max-tx-pwr-5g-mid-mcs7 = <0x38>;
+			wifi-max-tx-pwr-5g-high-mcs0 = <0x38>;
+			wifi-max-tx-pwr-5g-high-mcs7 = <0x38>;
+		};
+
+Please see the DTS binding documentation for more information.
```
[the remark this answers is not in the document]
Comment (richabp) [reply doc/nrf/drivers/wifi/native/nrf70.rst]: Can the link be added?

**Review by richabp — CHANGES_REQUESTED** [review]
Minor nits.

**Comment by krish2718** [comment]
> Only pending items it to link to the generated doxygen content.

As discussed with @gmarull , this needs more work and will be taken up later.

File: doc/nrf/drivers/wifi/nrf70_native.rst [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+Multiple calibrations and checks are implemented to ensure consistency across channels and devices.
+However, these values depend on PCB design, which may result in Error Vector Magnitude (EVM) and spectral mask failures.
+To avoid this problem, you can specify the power ceiling at which the EVM and spectral mask are met for a given PCB design.
+Additionally, build-time parameters are made available to drivers through the DTS overlay file.
+
+The following code snippet shows an example of the DTS overlay file.
+Note that the numbers used in this following example do not represent any particular PCB design or package type.
+
+You must replace these values with measurements obtained from transmitter testing on your own PCB designs.
+The values are represented in 0.25 dB increments.
```
Comment (krish2718) [inline doc/nrf/drivers/wifi/nrf70_native.rst:103]: This is not true, during upstreaming based on review comments, we have made this generic, so, removed the custom step size and used 1dBM, so, this and example values should be scaled down. Look at https://github.com/nrfconnect/sdk-zephyr/blob/main/drivers/wifi/nrfwifi/src/fmac_main.c#L497

**Comment by krish2718** [comment]
For some reason the doc build check has been disappeared, any ideas?

**Comment by krish2718** [comment]
> For some reason the doc build check has been disappeared, any ideas?

rebase seems to do the trick.

**Review by srkanordic — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Review by PerMac — DISMISSED** [review]
does this resolve any issue from the quarantine https://github.com/nrfconnect/sdk-nrf/blob/main/scripts/quarantine.yaml? If so, please also remove such entries from quarantine within this PR so we can let the CI verify the fixes. If not relevant, just ignore my comment and dismiss my request

**Comment by krish2718** [comment]
> does this resolve any issue from the quarantine https://github.com/nrfconnect/sdk-nrf/blob/main/scripts/quarantine.yaml? If so, please also remove such entries from quarantine within this PR so we can let the CI verify the fixes. If not relevant, just ignore my comment and dismiss my request

As commented https://github.com/nrfconnect/sdk-nrf/pull/18241#pullrequestreview-2401953783 Quarantine will be part of that PR.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 11]
```diff
@@ -1359,6 +1359,5 @@ Documentation
 
   * The Device configuration guides section and moved its contents to :ref:`ug_app_dev`.
   * The Advanced building procedures page and moved its contents to the :ref:`building` page.
-  * nRF70 Series support is upstreamed to Zephyr, hence the documentation is removed from the |NCS|.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:1362]: Please remove this file and add a comment in PR - https://github.com/nrfconnect/sdk-nrf/pull/18192.

File: CODEOWNERS [hunk 12]
```diff
@@ -98,6 +98,8 @@
 /doc/nrf/drivers/sensor_sim.rst           @nrfconnect/ncs-si-muffin-doc
 /doc/nrf/drivers/sensor_stub.rst          @nrfconnect/ncs-si-muffin-doc
 /doc/nrf/drivers/uart_ipc.rst             @nrfconnect/ncs-doc-leads
+/doc/nrf/drivers/wifi/                    @sachinthegreen @krish2718
+/doc/nrf/drivers/wifi.rst                 @sachinthegreen @krish2718
```
Comment (umapraseeda) [inline CODEOWNERS:102]:
```suggestion
/doc/nrf/drivers/wifi.rst                 @nrfconnect/ncs-wifi-doc
```

File: west.yml [hunk 13]
```diff
@@ -72,7 +72,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
```
Comment (rlubos) [inline west.yml:72]: Typo in the commit message :`mainfet: sdk-zephyr: `

**Review by rlubos — COMMENTED** [review]
Please update SHAs

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
@nrfconnect/ncs-co-drivers @nrfconnect/ncs-co-build-system Please review

**Review by anangl — APPROVED** [review]

**Pull request merged**